R/eml.R

```diff
@@ -1006,3 +1006,125 @@ reorder_pids <- function(pid_list, doc){
   ordered_pids <- pid_list[order(match(names(pid_list), entity_names))]
   return(ordered_pids)
 }
+
+#' Create an EML project section from a list of NSF award numbers
+#'
+#' This function takes a list of NSF award numbers and uses it to
+#' query the NSF API to get the award title, PIs, and coPIs. The
+#' return value is an EML project section. The function supports 1
+#' or more award numbers
+#'
+#' @param awards (list) A list of NSF award numbers as characters
+#' @param eml_version (char) EML version to use (2.1.1 or 2.2.0)
+#' @return project (emld) An EML project section
+#'
+#' @export
+#'
+#' @examples
+#' awards <- c("1203146", "1203473", "1603116")
+#'
+#' proj <- eml_nsf_to_project(awards, eml_version = "2.1.1")
+#'
+#' me <- list(individualName = list(givenName = "Jeanette", surName = "Clark"))
+#'
+#' doc <- list(packageId = "id", system = "system",
+#'            dataset = list(title = "A Mimimal Valid EML Dataset",
+#'                           creator = me,
+#'                           contact = me))
+#'
+#' doc$dataset$project <- proj
+#'
+#' EML::eml_validate(doc)
+#'
+eml_nsf_to_project <- function(awards, eml_version = "2.1"){
+
+  stopifnot(is.character(awards))
+  stopifnot(eml_version %in% c("2.1", "2.1.1", "2.2", "2.2.0"))
+
+  award_nums <- awards
+
+  result <- lapply(award_nums, function(x){
+    url <- paste0("https://api.nsf.gov/services/v1/awards.json?id=", x ,"&printFields=coPDPI,pdPIName,title")
+
+    t <- jsonlite::fromJSON(url)
```

Comment:
Any reason this is named `t`?  Doesn't need to be changed just wondering

Reply:
not really, just a short placeholder name. `response` would be a natural name but the return value for `fromJSON` already has `response` in it